api/db/populate-friendships.js

```diff
@@ -0,0 +1,85 @@
+const { getRandomSubset, getRandomDate, createPrompt } = require('./utils');
+
+const User = require('../models/User');
+const Friendship = require('../models/Friendship');
+
+let displayLogs;
+
+const createFriendship = async (friendshipParams) => {
+  const friendship = new Friendship({
+    requestant: friendshipParams.requestant,
+    receivant: friendshipParams.receivant,
+    accepted: friendshipParams.accepted,
+    acceptedAt: friendshipParams.acceptedAt,
+    lastMessageAt: friendshipParams.lastMessageAt,
+  });
+  await friendship.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New friendship created: ${friendship}`);
+  }
+  return friendship;
+};
+
+const createFriendships = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating friendships collection...');
+  displayLogs = verbose;
+
+  let usersList = await User.distinct('_id');
+  const friendshipsArray = await Friendship.find();
+  let recordCount = friendshipsArray.length;
+
+  let proceed = true;
+  if (recordCount > 0) {
+    proceed = await createPrompt(
+      'Some users already have friends. Would you like to add friends to users that already contain friends?'
+    );
+  }
+
+  for (const user of usersList) {
+    const containFriends = friendshipsArray.some(
+      (friendship) =>
+        friendship.requestant.toString() === user.toString() ||
+        friendship.receivant.toString() === user.toString()
+    );
+    if (!proceed && containFriends) continue;
+
+    // split current user from all others
+    const tmpList = usersList.filter((userValue) => userValue != user);
+    // create subset userList from remaining usersList
+    const usersSubset = getRandomSubset(tmpList, tmpList.length);
+
+    for (const receivant of usersSubset) {
+      const alreadFriends = friendshipsArray.filter((friendship) => {
+        return (
+          (friendship.requestant.toString() === user.toString() &&
+            friendship.receivant.toString() === receivant.toString()) ||
+          (friendship.requestant.toString() === receivant.toString() &&
+            friendship.receivant.toString() === user.toString())
+        );
+      });
+      if (alreadFriends.length > 0) continue;
+
+      const now = new Date();
+      const acceptedAt = getRandomDate(
+        new Date(2020, 4, 4),
+        new Date(now.getTime() - 60000)
+      );
+      const friendshipParams = {
+        requestant: user,
+        receivant,
+        accepted: Math.random() > 0.1 ? true : false,
```

Comment:
You can remove the ` ? true : false` part from here, as `Math.random() > 0.1` will already evaluate to `true` or `false`. Also, should we make it a bit more probable for a friendship request to not be accepted? Something like `Math.random >= 0.2` or `Math.random >= 0.25`?